clang/lib/Basic/Attributes.cpp

```diff
@@ -153,12 +155,33 @@ std::string AttributeCommonInfo::getNormalizedFullName() const {
       normalizeName(getAttrName(), getScopeName(), getSyntax()));
 }
 
+const llvm::StringMap<AttributeCommonInfo::Scope> ScopeMap = {
```

Comment:
Avoid static constructor? Just make this a sorted list and use lower_bound or one of the other search functions?